faiss/CMakeLists.txt

```diff
@@ -149,7 +149,12 @@ if(NOT WIN32)
 endif()
 
 if(FAISS_OPT_LEVEL STREQUAL "avx2")
-  target_compile_options(faiss PRIVATE $<$<COMPILE_LANGUAGE:CXX>:-mavx2 -mfma -mf16c -mpopcnt>)
+  if(NOT WIN32)
+    target_compile_options(faiss PRIVATE $<$<COMPILE_LANGUAGE:CXX>:-mavx2 -mfma -mf16c -mpopcnt>)
+  else()
+    # MSVC enables FMA with /arch:AVX2; no separate flags for F16C, POPCNT
```

Comment:
If you could add a link to MSVC's docs backing that, that would be awesome.

Reply:
The problem is that windows doesn't particularly document the differences to GNU-land, but rather just what it does.

I've collected some references, but most of those insights ultimately came from somewhere on stackoverflow, like [here](https://stackoverflow.com/a/50829580/2965879).